### PR TITLE
[tests] Add infrastructure to run tests on Windows that need a remote Mac. Fixes #13924.

### LINKED ISSUE #13924: CI: figure out CI testing on Windows
We need to figure out how to run tests from Windows (using a remote Mac).

List of tests that would be nice to implement: https://github.com/xamarin/xamarin-macios/issues/19926

From the XVS team:

> We have Windows and Mac bots on the same network, and then the pipeline holds a Mac and passes its info to the Windows bot so it can use that information for the pairing process
> For instance [[NEW] MAUI Insertion Gate Tests - team.xamarin.maui-net7-sr3-d17_5-33325.28.jodick.230125.183736-418 - Pipelines (visualstudio.com)](https://devdiv.visualstudio.com/DevDiv/_releaseProgress?_a=release-pipeline-progress&releaseId=1424533)
> The pool VSEng-VSMac-Xamarin-Shared has both Windows and Mac bots
> Btw, we are doing this on a yaml pipeline too: [VisualStudio.Apex/reserve-mac.yaml at main · xamarin/VisualStudio.Apex (github.com)](https://github.com/xamarin/VisualStudio.Apex/blob/main/build/stages/reserve-mac.yaml) and [VisualStudio.Apex/build.yml at main · xamarin/VisualStudio.Apex (github.com)](https://github.com/xamarin/VisualStudio.Apex/blob/main/build/stages/build.yml#L14)

### PR DESCRIPTION
**Pull request by rolfbjarne**
Also add a single test that builds a simple project remotely.

This PR might be best reviewed commit-by-commit.

Fixes https://github.com/xamarin/xamarin-macios/issues/13924.

### CODE REVIEW

File: tests/common/DotNet.cs [hunk 1]
```diff
@@ -113,7 +113,7 @@ public static ExecutionResult AssertNew (string outputDirectory, string template
 			return new ExecutionResult (output, output, rv.ExitCode);
 		}
 
-		public static ExecutionResult Execute (string verb, string project, Dictionary<string, string>? properties, bool assert_success = true, string? target = null, bool? msbuildParallelism = null)
+		public static ExecutionResult Execute (string verb, string project, Dictionary<string, string>? properties, bool assert_success = true, string? target = null, bool? msbuildParallelism = null, TimeSpan? timeout = null, bool? quiet = null)
```
Comment (mandel-macaque) [inline tests/common/DotNet.cs:116]: Why a bool? and not default to false? I hate having to deal with nullability and bools.

File: tools/devops/automation/scripts/RemoteMac.psm1 [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        $CommandArguments
+    )
+
+    $cmd = "ssh -v -i `"$Env:ID_RSA_PATH`" -o StrictHostKeyChecking=no `"$($RemoteUserName)@$($RemoteHost)`" $CommandArguments"
+    Write-Host "Executing: $cmd"
+
+    ssh -v -i "$Env:ID_RSA_PATH" -o StrictHostKeyChecking=no "$($RemoteUserName)@$($RemoteHost)" @CommandArguments
+
+    if ($LastExitCode -ne 0) {
+        throw [System.Exception]::new("Failed to execute ssh command '$cmd', exit code: $LastExitCode")
```
Comment (mandel-macaque) [inline tools/devops/automation/scripts/RemoteMac.psm1:27]: For the lazy, you can do a simple throw with a string.

**Review by mandel-macaque — APPROVED** [review]
minor comments that should not block.

File: tools/devops/automation/scripts/RemoteMac.psm1 (hunk 2 above)
Comment (rolfbjarne) [reply tools/devops/automation/scripts/RemoteMac.psm1:27]: Nice to know!

File: tests/common/DotNet.cs (hunk 1 above)
Comment (rolfbjarne) [reply tests/common/DotNet.cs]: Fixed.

**Review by dalexsoto — APPROVED** [review]

**Pull request merged**